contracts/misc/BountyPayout.sol

```diff
@@ -0,0 +1,182 @@
+      CHILD_SKILL_INDEX,
+      1, // Root domain funding pot is always 1
+      payment.fundingPotId,
+      _amount,
+      leapAddr
+    );
+    colony.finalizePayment(PERMISSION_DOMAIN_ID, CHILD_SKILL_INDEX, paymentId);
+
+    // Claim payout on behalf of the recipient
+    colony.claimPayment(paymentId, leapAddr);
+    return paymentId;
+  }
+
+  function _payout(
+    address payable _gardenerAddr,
+    uint256 _gardenerDaiAmount,
+    address payable _workerAddr,
+    uint256 _workerDaiAmount,
+    address payable _reviewerAddr,
+    uint256 _reviewerDaiAmount,
+    bytes32 _bountyId
+  ) internal  {
+
+    IERC20 dai = IERC20(daiAddr);
+
+    // handle worker
+    uint256 paymentId = makePayment(_gardenerAddr, _gardenerDaiAmount);
+    dai.transferFrom(payerAddr, _gardenerAddr, _gardenerDaiAmount);
+    emit Payout(_bountyId, PayoutType.Gardener, _gardenerAddr, _gardenerDaiAmount, paymentId);
+
+    // handle worker
+    if (_workerDaiAmount > 0) {
+      paymentId = makePayment(_workerAddr, _workerDaiAmount);
+      dai.transferFrom(payerAddr, _workerAddr, _workerDaiAmount);
+      emit Payout(_bountyId, PayoutType.Worker, _workerAddr, _workerDaiAmount, paymentId);
+    }
+
+    // handle reviewer
+    if (_reviewerDaiAmount > 0) {
+      paymentId = makePayment(_reviewerAddr, _reviewerDaiAmount);
+      dai.transferFrom(payerAddr, _reviewerAddr, _reviewerDaiAmount);
+      emit Payout(_bountyId, PayoutType.Reviewer, _reviewerAddr, _reviewerDaiAmount, paymentId);
+    }
+  }
+
+ /**
+  * Pays out a bounty to the different roles of a bounty
+  *
+  * @dev This contract should have enough allowance of daiAddr from payerAddr
+  * @dev This colony contract should have enough LEAP in its funding pot
+  * @param _gardenerAddr gardener wallet address
+  * @param _gardenerDaiAmount DAI amount to pay gardner
+  * @param _workerAddr worker wallet address
+  * @param _workerDaiAmount DAI amount to pay worker
+  * @param _reviewerAddr reviewer wallet address
+  * @param _reviewerDaiAmount DAI amount to pay reviewer
+  */
+  function payout(
+    address payable _gardenerAddr,
+    uint256 _gardenerDaiAmount,
+    address payable _workerAddr,
+    uint256 _workerDaiAmount,
+    address payable _reviewerAddr,
+    uint256 _reviewerDaiAmount,
+    bytes32 _bountyId
+  ) public onlyPayer {
+    _payout(
+      _gardenerAddr,
+      _gardenerDaiAmount,
+      _workerAddr,
+      _workerDaiAmount,
+      _reviewerAddr,
+      _reviewerDaiAmount,
+      _bountyId
+    );
+  }
+
+  function payoutNoWorker(
+    address payable _gardenerAddr,
+    uint256 _gardenerDaiAmount,
+    address payable _reviewerAddr,
+    uint256 _reviewerDaiAmount,
+    bytes32 _bountyId
+  ) public onlyPayer {
+    _payout(
+      _gardenerAddr,
+      _gardenerDaiAmount,
+      _reviewerAddr,
+      0,
+      _reviewerAddr,
+      _reviewerDaiAmount,
+      _bountyId
+    );
+  }
+
+  function payoutNoReviewer(
```

Comment:
👍

Reply:
sometimes we need just one payout (e.g. invoice for Ops). I guess we can call `payoutNoReviewer` with 0 for gardener share?

Reply:
> with 0 for gardener share?
i would put 0 for workerShare.

Reply:
semantically it doesn't make sense — worker share should always be non-zero (otherwise what kind of work is done on the bounty at all). I had no chance to review this change in the code, btw.. not great

Reply:
no need to change anything now, of course

Reply:
> I had no chance to review this change in the code, btw.. not great

I found a bug/shortcoming after you approved already. should have requested re-review.

Reply:
> otherwise what kind of work is done on the bounty at all

yeah, I thought without the gardener bounty couldn't exist at all...